l10n_eu_nace/wizards/nace_import.py

```diff
@@ -0,0 +1,274 @@
+                name = 'res.partner.nace,' + field
+                query = [('res_id', '=', nace.id),
+                         ('type', '=', 'model'),
+                         ('name', '=', name),
+                         ('lang', '=', lang.code)]
+                translation = translation_model.search(query)
+                data = {
+                    'value': value,
+                    'state': 'translated'
+                }
+                if translation:
+                    translation_model.write(data)
+                    # TODO  2019-01-18: dangerous code, write in ir_translation
+                    self.env.cr.execute(
+                        "UPDATE ir_translation SET value = "
+                        "%s WHERE id = %s", (
+                            value.replace("'", "''"), translation.id))
+                else:
+                    data['res_id'] = nace.id
+                    data['type'] = 'model'
+                    data['name'] = name
+                    data['lang'] = lang.code
+                    data['module'] = 'l10n_eu_nace'
+                    translation_model.create(data)
+
+    @api.multi
+    def run_import(self):
+        nace_model = self.env['res.partner.nace'].\
+            with_context(defer_parent_store_computation=True)
+        lang_model = self.env['res.lang']
+        # Available lang list
+        langs = lang_model.search(
+            [('code', 'in', list(self._available_langs.keys())),
+             ('active', '=', True)])
+        # All current NACEs, delete if not found above
+        naces_to_delete = nace_model.search([])
+        # Download NACEs in english, create or update
+        logger.info('Import NACE Rev.2 English')
+        xmlcontent = self._download_nace('EN')
+        # pylint: disable=c-extension-no-member
+        utf8_parser = etree.XMLParser(encoding='utf-8')
+        dom = etree.fromstring(xmlcontent.encode('utf-8'), parser=utf8_parser)
+        for node in dom.iter('Item'):
+            logger.debug(
+                'Reading level=%s, code=%s', node.get('idLevel', 'N/A'),
+                node.get('id', 'N/A'))
+            nace = self.create_or_update_nace(node)
+            if nace and nace in naces_to_delete:
+                naces_to_delete -= nace
+        # Download NACEs in other languages, translate them
+        for lang in langs:
+            logger.info('Import NACE Rev.2 %s', lang.code)
+            nace_lang = self._available_langs[lang.code]
+            xmlcontent = self._download_nace(nace_lang)
+            dom = etree.fromstring(
+                xmlcontent.encode('utf-8'), parser=utf8_parser)
+            for node in dom.iter('Item'):
+                logger.debug(
+                    'Reading lang=%s, level=%s, code=%s', nace_lang,
+                    node.get('idLevel', 'N/A'), node.get('id', 'N/A'))
+                self.translate_nace(node, lang)
+        # Delete obsolete NACEs
+        if naces_to_delete:
```

Comment:
The wizard should have a bool option for this:

```suggestion
        if self.delete_obsolete_records and naces_to_delete:
```